scripts/qa_crowbarsetup.sh

```diff
@@ -1476,34 +1476,30 @@ function proposal_increment_int()
     proposal_modify_value "$1" "$2" "$3" "$4" "+="
 }
 
-function enable_ssl_for_keystone()
+function enable_ssl_generic()
 {
-    echo "Enabling SSL for keystone"
-    proposal_set_value keystone default "['attributes']['keystone']['api']['protocol']" "'https'"
-}
-
-function enable_ssl_for_glance()
-{
-    echo "Enabling SSL for glance"
-    proposal_set_value glance default "['attributes']['glance']['api']['protocol']" "'https'"
-}
-
-function enable_ssl_for_nova()
-{
-    echo "Enabling SSL for nova"
-    proposal_set_value nova default "['attributes']['nova']['api']['protocol']" "'https'"
-    proposal_set_value nova default "['attributes']['nova']['glance_ssl_no_verify']" true
-    proposal_set_value nova default "['attributes']['nova']['novnc']['ssl_enabled']" true
-}
-
-
-function enable_ssl_for_nova_dashboard()
-{
-    echo "Enabling SSL for nova_dashboard"
-    proposal_set_value nova_dashboard default "['attributes']['nova_dashboard']['apache']['use_https']" true
-    proposal_set_value nova_dashboard default "['attributes']['nova_dashboard']['apache']['use_http']" false
-    proposal_set_value nova_dashboard default "['attributes']['nova_dashboard']['apache']['redirect_to_https']" false
-    proposal_set_value nova_dashboard default "['attributes']['nova_dashboard']['ssl_no_verify']" true
+    local service=$1
+    echo "Enabling SSL for $service"
+    local p="proposal_set_value $service default"
+    local a="['attributes']['$service']"
+    case $service in
+        swift)
+            $p "$a['ssl']['enabled']" true
+        ;;
+        nova)
+            $p "$a['ssl']['enabled']" true
+            $p "$a['novnc']['ssl']['enabled']" true
+        ;;
+        nova_dashboard)
+            $p "$a['apache']['ssl']" true
+            return
+        ;;
+        *)
```

Comment:
why not `keystone|glance|neutron)` here? although not really necessary codewise, but it would make it a bit more clear, and why was there no `enable_ssl_for_neutron` before?